src/buf.ts

```diff
@@ -36,23 +36,33 @@ export async function getBuf(version: string): Promise<string|Error> {
       return downloadURL
   }
 
+  let cacheDir = "";
   core.info(`Downloading buf version "${version}" from ${downloadURL}`);
-  const downloadPath = await tc.downloadTool(downloadURL);
-  core.info(`Successfully downloaded buf version "${version}" from ${downloadURL}`);
+  if (downloadURL.endsWith('.tar.gz')){
+    const downloadPath = await tc.downloadTool(downloadURL);
+    core.info(`Successfully downloaded buf version "${version}" from ${downloadURL}`);
 
-  core.info('Extracting buf...');
-  const extractPath = await tc.extractTar(downloadPath);
-  core.info(`Successfully extracted buf to ${extractPath}`);
+    core.info('Extracting buf...');
+    const extractPath = await tc.extractTar(downloadPath);
+    core.info(`Successfully extracted buf to ${extractPath}`);
 
-  core.info('Adding buf to the cache...');
-  const cacheDir = await tc.cacheDir(
-    path.join(extractPath, 'buf'),
-    'buf',
-    version,
-    os.arch()
-  );
-  core.info(`Successfully cached buf to ${cacheDir}`);
+    core.info('Adding buf to the cache...');
+    cacheDir = await tc.cacheDir(
+      path.join(extractPath, 'buf'),
+      'buf',
+      version,
+      os.arch()
+    );
+  } else {
+    // For Windows, we only download the .exe for `buf` CLI becasue we do not create `.tar.gz`
+    // bundles for Windows releases.
+    const downloadPath = await tc.downloadTool(downloadURL, 'C:\\Users\\runnderadmin\\buf.exe');
```

Comment:
Is this `C:\\Users\\runnderadmin\\buf.exe` standard across all GitHub runners? Can we link to documentation that mentions this in-line?

Reply:
Should it be `runneradmin` 🤔?

Reply:
It should be. and Yeah, I can link documentation. Interestingly, this also worked, because it just created the directory, lol. But it's better to use a known directory.